### PR TITLE
fix(workflow): Change "View in Discover" link to include `interval`

### PR DESCRIPTION
**Pull request by billyvg**
Previously it was using the `timeWindow` alert rule setting as the query
period, which is wrong. Update to set `interval` for the events-stats
request to use `timeWindow`, and set the start/end dates to incident start
-> end (or now)

Requires https://github.com/getsentry/sentry/pull/17119

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx [hunk 1]
```diff
@@ -57,11 +58,17 @@ export default class DetailsBody extends React.Component<Props> {
         .filter(({slug}) => incident.projects.includes(slug))
         .map(({id}) => Number(id)),
       version: 2 as const,
-      range: `${incident.alertRule.timeWindow}m`,
+      start: incident.dateStarted,
+      end: incident.dateClosed ?? getUtcDateString(new Date()),
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/alerts/details/body.tsx]: This start/end range could be bad if it started very recently (e.g. 1 minute ago).

Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: @billyvg why is that bad? does that produce undesirable results?

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx [hunk 2]
```diff
@@ -57,11 +58,17 @@ export default class DetailsBody extends React.Component<Props> {
...
     };
 
     const discoverView = EventView.fromSavedQuery(discoverQuery);
-    return discoverView.getResultsViewUrlTarget(orgId);
+    const {query, ...toObject} = discoverView.getResultsViewUrlTarget(orgId);
+
+    return {
+      query: {...query, interval: `${incident.alertRule.timeWindow}m`},
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/alerts/details/body.tsx]: @billyvg I don't think this `interval` query string is actually being passed anywhere. For example hardcode this to `3m`. None of the API endpoint calls is making use of this.

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: Bad as in you could have a start-end range of 1 minute with a time window of 1 minute, so you only chart 1 data point

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: It needs this https://github.com/getsentry/sentry/pull/17119

Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: Ah gotcha 👍

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: Oh I see.

Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: For that edge case, we could set `start` to be earlier; like 30minutes or so back.

Maybe this function could help in detecting tight chart windows. https://github.com/getsentry/sentry/blob/9a802f3d69f8bea59c834aaf7762861be90e1603/src/sentry/static/sentry/app/components/charts/utils.tsx#L65

Comment (billyvg) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: We're gonna do (start - timewindow - 20% of diff of start/end or now)

**Review by dashed — APPROVED** [review]
Looks good to me 👍

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: Yea that works. 👍

File: src/sentry/static/sentry/app/views/alerts/details/body.tsx [hunk 3]
```diff
@@ -43,6 +45,16 @@ export default class DetailsBody extends React.Component<Props> {
...
 
+    const timeWindowString = `${incident.alertRule.timeWindow}m`;
+    const timeWindowInMs = intervalToMilliseconds(timeWindowString);
+    const started = moment(incident.dateStarted);
+    const end = incident.dateClosed ?? getUtcDateString(new Date());
+
+    // We want the discover chart to start at "dateStarted" - "timeWindow" - "20%"
+    const start = started
+      .subtract(timeWindowInMs, 'ms')
+      .subtract(moment(end).diff(started, 'ms') * 0.2, 'ms');
```
Comment (adhiraj) [inline src/sentry/static/sentry/app/views/alerts/details/body.tsx]: this should be 20% of (end - (started - timeWindow))

Comment (dashed) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: @adhiraj that could be `0` if `end - start == timeWindow`

Comment (billyvg) [reply src/sentry/static/sentry/app/views/alerts/details/body.tsx]: Updated

**Pull request merged**